picasso/src/main/java/com/squareup/picasso3/RequestCreator.kt

```diff
@@ -0,0 +1,663 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.picasso3
+
+import android.app.Notification
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.view.Gravity
+import android.widget.ImageView
+import android.widget.RemoteViews
+import androidx.annotation.DimenRes
+import androidx.annotation.DrawableRes
+import androidx.annotation.IdRes
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.ContextCompat
+import com.squareup.picasso3.BitmapHunter.Companion.forRequest
+import com.squareup.picasso3.MemoryPolicy.Companion.shouldReadFromMemoryCache
+import com.squareup.picasso3.MemoryPolicy.Companion.shouldWriteToMemoryCache
+import com.squareup.picasso3.Picasso.LoadedFrom
+import com.squareup.picasso3.PicassoDrawable.setPlaceholder
+import com.squareup.picasso3.PicassoDrawable.setResult
+import com.squareup.picasso3.RemoteViewsAction.AppWidgetAction
+import com.squareup.picasso3.RemoteViewsAction.NotificationAction
+import com.squareup.picasso3.RemoteViewsAction.RemoteViewsTarget
+import com.squareup.picasso3.Utils.OWNER_MAIN
+import com.squareup.picasso3.Utils.VERB_COMPLETED
+import com.squareup.picasso3.Utils.checkMain
+import com.squareup.picasso3.Utils.checkNotMain
+import com.squareup.picasso3.Utils.log
+import java.io.IOException
+import java.util.concurrent.atomic.AtomicInteger
+
+/** Fluent API for building an image download request.  */
+open class RequestCreator {
```

Comment:
Had to mark this as open for now to satisfy mockito 😞

Reply:
I'll fix that in a follow-up.